src/components/DatePickerInline/YearPicker.tsx

```diff
@@ -1,39 +1,39 @@
-import { useRef, useEffect, memo, useMemo } from 'react';
+import { useRef, useEffect, memo, useMemo, useCallback } from 'react';
 import { StyleSheet } from 'react-native';
 import { useComponentStyles, useMolecules } from '../../hooks';
 import type { FlatListRef } from '../FlatList';
-import { range } from '../../utils/dateTimePicker';
+import { getYearRange } from '../../utils';
 
 const ITEM_HEIGHT = 62;
 
 export default function YearPicker({
     selectedYear,
     selectingYear,
-    onPressYear,
+    onChange,
     startYear,
     endYear,
 }: {
     selectedYear: number | undefined;
     selectingYear: boolean;
-    onPressYear: (year: number) => any;
-    startYear: number;
-    endYear: number;
+    onChange: (year: number, type: 'month' | 'year') => any;
+    startYear?: number;
+    endYear?: number;
 }) {
-    const { FlatList, View } = useMolecules();
+    const { FlatList, View, HorizontalDivider } = useMolecules();
     const yearPickerStyles = useComponentStyles('DatePicker_YearPicker');
     const flatList = useRef<FlatListRef<number> | null>(null);
-    const years = range(isNaN(startYear) ? 1800 : startYear, isNaN(endYear) ? 2200 : endYear);
+    const years = getYearRange(startYear, endYear);
```

Comment:
memoize `getYearRange(startYear, endYear);`

Reply:
done in new PR @shashank1010